pandas/core/internals/concat.py

```diff
@@ -92,6 +101,93 @@ def _concatenate_array_managers(
         return ArrayManager(arrays, [axes[1], axes[0]], verify_integrity=False)
 
 
+def concat_arrays(to_concat: List[Any]) -> ArrayLike:
+    """
+    Alternative for concat_compat but specialized for use in the ArrayManager.
+
+    Differences: only deals with 1D arrays (no axis keyword), assumes
+    ensure_wrapped_if_datetimelike and does not skip empty arrays to determine
+    the dtype.
+    In addition ensures that all NullArrayProxies get replaced with actual
+    arrays.
+
+    Parameters
+    ----------
+    to_concat : list of arrays
```

Comment:
the annotation is List[Any].  can that be made more specific?

Reply:
In principle the more specific annotation would be `List[Union[np.ndarray, ExtensionArray, NullArrayProxy]]`. But, iwhen adding that, that gives a bunch of errors down the line because mypy cannot figure out that we are eg converting all input to EAs before passing to `_concat_same_type`. 
So unless if you have specific suggestions how to solve this, I'd rather leave it less specified (for example `concat_compat` is also not typed).

Reply:
i think the practice in this case is to not annotate rather than use Any, cc @simonjayhawkins

Reply:
But `List[Any]` at least indicates it needs to receive a list, which can be better than no typing at all?

Reply:
im hoping @simonjayhawkins will weigh in here, but my understanding of the rule of thumb is that `List[Any]` denotes "this cannot be further narrowed down" whereas `List` denotes "its a list and the annotation is incomplete; contributions welcome"

Reply:
Ah, I thought you meant to have no annoation at all (like `concat_compat`). I can certainly make it `List` instead of `List[Any]`

Reply:
I _think_ that's the general consensus at the moment until we add `disallow_any_generics` #30539 which will force us to add type parameters.